app/src/main/java/org/android/go/sopt/presentation/auth/SignUpActivity.kt

```diff
@@ -59,4 +61,31 @@ class SignUpActivity : BindingActivity<ActivitySignUpBinding>(R.layout.activity_
             it.hideKeyboard()
         }
     }
+
+    private fun validInput() {
+        viewModel.isIdValid.observe(this) {
+            Log.d("signUp", viewModel.isIdValid.value.toString())
+            if (viewModel.isIdValid.value == false) {
+                binding.edtSignupIdMain.error = "아이디 형식이 잘못 되었어요 :("
+            } else {
+                binding.edtSignupIdMain.error = null
+                binding.edtSignupIdMain.isErrorEnabled = false
+            }
+        }
+        viewModel.isPwdValid.observe(this) {
+            Log.d("signUp", viewModel.isPwdValid.value.toString())
+            if (viewModel.isPwdValid.value == false) {
+                binding.edtSignupPwdMain.error = "비밀번호 형식이 잘못 되었어요 :("
+            } else {
+                binding.edtSignupPwdMain.error = null
+                binding.edtSignupPwdMain.isErrorEnabled = false
```

Comment:
```suggestion
        viewModel.isPwdValid.observe(this) {
            if (!it) {
                binding.edtSignupPwdMain.error = getString(R.string.invalid_id)
            } else {
                binding.edtSignupPwdMain.error = null
                binding.edtSignupPwdMain.isErrorEnabled = false
```

it을 활용해도 좋을 것 같습니다!
그리고 문자열을 resource로 관리하면 추후 코드 변경 없이 유지보수가 가능할 것 같아요!

Reply:
감사함미다 좀더 알아보기 쉽게 직관적인 네이밍으로 it 대신 isPwdValid으로 변경했습니당 ! 스트링 추출도 하겠습니다 ㅎㅎ